lib/db/db.rb

```diff
@@ -1,69 +1,27 @@
 class DB
 
-  def initialize(reader, writer)
-    @reader = reader
-    @writer = writer
-  end
-
   def size
-    all.size
   end
 
   def all
-    reader.read
   end
 
   def at(index)
-    all.fetch(index, {})
   end
 
   def add(contact)
-    writer.write(added_a(contact))
   end
 
   def update(contact)
-    writer.write(updated_with(add_id_to(contact)))
   end
 
   def search(term)
-    all.select { |contact| contact.any? {|key, val| val.to_s.include?(term)} }
   end
 
   def index_of_id(id)
-    all.index { |contact| contact["id"] == id }
-  end
-
-  private
-
-  attr_reader :reader, :writer
-
-  def invalid?(index)
-    index.nil? ? true : all[index].nil?
-  end
-
-  def add_id_to(contact)
-    contact.has_key?("id") ? contact : append_id(contact, id_of_new)
-  end
-
-  def append_id(contact, id)
-    contact_with_id       = {}
-    contact_with_id["id"] = id
-    contact_with_id.merge!(contact)
-  end
-
-  def id_of_new
-    size == 0 ? 1 : at(size - 1)["id"] + 1
-  end
-
-  def added_a(contact)
-    all << add_id_to(contact)
   end
 
-  def updated_with(contact)
-    index    = index_of_id(contact["id"])
-    contacts = all
-    invalid?(index) ? contacts << contact : contacts[index] = contact
-    contacts
+  def close
   end
 
-end
\ No newline at end of file
+end
```

Comment:
This file does have any meaning or provide any value and therefore should be deleted.